src/python/CRABClient/CrabRestInterface.py

```diff
@@ -182,6 +184,8 @@ def makeRequest(self, uri=None, data=None, verb='GET'):
             command += '/cvmfs/cms.cern.ch/cmsmon/gocurl -verbose 2 -method {0}'.format(verb)
             command += ' -header "User-Agent: %s/%s"' % (self['userAgent'], self['version'])
             command += ' -header "Accept: */*"'
+            if self['Content-type']:
+                command += ' -header "Content-type: %s"' % self['Content-type']
             if verb in ['POST', 'PUT']:
                 command += ' -header "Content-Type: application/x-www-form-urlencoded"'
```

Comment:
```suggestion
            if self['Content-type']:
                command += ' -header "Content-type: %s"' % self['Content-type']
            else:
                command += ' -header "Content-Type: application/x-www-form-urlencoded"'
```
When body (`--data`) is supplied in curl, default `Content-Type` is `application/x-www-form-urlencoded`.

Reply:
those lines are under the `useGoCurl` which is 1) little tested 2) not used atm 3) . So I did not pay much attention, sorry. Nor tested (bad, I know). I am not  sure that the difference eventually is since  `default Content default Content-Type is application/x-www-form-urlencoded` seems to apply to all verbs, IIUC. So in a way the original lines were likely useless ? Showing master branch here:
https://github.com/dmwm/CRABClient/blob/0bc1ad01ad739a1c6219a40c0cbd4882bc1ed3f4/src/python/CRABClient/CrabRestInterface.py#L185-L186

Shouldn't I then use same suggested change as above for the code that follows the `else:` and applies when not using GoCurl ?

Reply:
No need. Curl is working fine because it is [default behavior from curl when you supply `--data` param](https://everything.curl.dev/http/post/content-type), also apply to GET I tested in my local machine. However I am not sure about gocurl, so I suggested as above.

Reply:
OK. Let's do same way for `GoCurl` and `curl`. If we ever come to wanting to use `GoCurl` we will see. Most likely Valentin did things correctly and behaves exactly same as `curl`.